### PR TITLE
Fix pt2 dashboard passrate calculation regression

### PR DESCRIPTION
**Pull request by huydhn**
The regression comes from https://github.com/pytorch/test-infra/pull/6006 where I changed `accuracy_results LEFT JOIN performance_results` to `performance_results LEFT JOIN accuracy_results` to accommodate Apple MPS eager benchmark.  The swapped join wrongly returned `model_fail_to_load` and `eager_fail_to_run` models that shouldn't be included in the pass rate calculation because they are not something wrong with pt2.

### Testing

https://torchci-git-fork-huydhn-fix-pt2-dashboard-p-e92d66-fbopensource.vercel.app/benchmark/compilers?dashboard=torchinductor&startTime=Fri%2C%2019%20Jul%202024%2000%3A29%3A48%20GMT&stopTime=Wed%2C%2015%20Jan%202025%2001%3A29%3A48%20GMT&granularity=week&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=1dab79470dbecef79ba4c7d4308d8a181091e58e&rBranch=main&rCommit=b732b52f1e4378f8486ceb5e7026be3321c2651c

* Before https://hud.pytorch.org/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Thu%2C%2018%20Jul%202024%2023%3A36%3A11%20GMT&stopTime=Wed%2C%2015%20Jan%202025%2000%3A36%3A11%20GMT&granularity=week&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=b732b52f1e4378f8486ceb5e7026be3321c2651c&rBranch=main&rCommit=b732b52f1e4378f8486ceb5e7026be3321c2651c

* After https://torchci-git-fork-huydhn-fix-pt2-dashboard-p-e92d66-fbopensource.vercel.app/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Thu%2C%2018%20Jul%202024%2023%3A36%3A11%20GMT&stopTime=Wed%2C%2015%20Jan%202025%2000%3A36%3A11%20GMT&granularity=week&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=b732b52f1e4378f8486ceb5e7026be3321c2651c&rBranch=main&rCommit=b732b52f1e4378f8486ceb5e7026be3321c2651chttps%3A%2F%2Ftorchci-git-fork-huydhn-fix-pt2-dashboard-p-e92d66-fbopensource.vercel.app%2F

### CODE REVIEW

**Comment by huydhn** [comment]
Just to confirm that the fix works correctly, I have a dummy PR to revert the change https://github.com/pytorch/test-infra/pull/6171 and observe the same number there https://torchci-git-fork-huydhn-revert-dd19e346ad18-2ae2b5-fbopensource.vercel.app/benchmark/compilers?dashboard=torchinductor&startTime=Fri%2C%2019%20Jul%202024%2000%3A50%3A20%20GMT&stopTime=Wed%2C%2015%20Jan%202025%2001%3A50%3A20%20GMT&granularity=week&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=1dab79470dbecef79ba4c7d4308d8a181091e58e&rBranch=main&rCommit=b732b52f1e4378f8486ceb5e7026be3321c2651c

File: torchci/clickhouse_queries/compilers_benchmark_performance/query.sql [hunk 1]
```diff
@@ -144,6 +142,62 @@ results AS (
         LEFT JOIN accuracy_results ON performance_results.name = accuracy_results.name
         AND performance_results.replaced_filename = accuracy_results.replaced_filename
         AND performance_results.workflow_id = accuracy_results.workflow_id
+    WHERE
+        accuracy != 'model_fail_to_load'
+        AND accuracy != 'eager_fail_to_run'
+),
+-- This is to accommodate cases where only accuracy results are available, i.e. export
```
Comment (huydhn) [inline torchci/clickhouse_queries/compilers_benchmark_performance/query.sql:149]: I discover another regression here where export results were missing from the dashboard.  That's a special case where there is only accuracy result for `torch.export`.

**Review by zou3519 — APPROVED** [review]

**Review by shunting314 — APPROVED** [review]
